src/modules/ekf2/Utility/PreFlightChecker.cpp

```diff
@@ -92,16 +93,16 @@ bool PreFlightChecker::preFlightCheckHorizVelFailed(const ekf2_innovations_s &in
 	return has_failed;
 }
 
-bool PreFlightChecker::preFlightCheckVertVelFailed(const ekf2_innovations_s &innov, const float alpha)
+bool PreFlightChecker::preFlightCheckVertVelFailed(const estimator_innovations_s &innov, const float alpha)
 {
-	const float vel_d_innov = innov.vel_pos_innov[2];
+	const float vel_d_innov = fmaxf(fabsf(innov.gps_vvel), fabs(innov.ev_vvel));
```

Comment:
This seems a bit sketchy. What if you have terrible GPS vel, but good vision vel? Should this be checking for the minimum instead?

Reply:
I also was thinking about this. Same applies for velocity and position innovation checks during missions in Commander.

Reply:
The issue is that innovations are published as zero if there is no fusion happening. So we would need to reject the zero case. For the moment, taking the max is the conservative approach.

Reply:
Not convinced that this is the correct approach. It seems to me that this will just always fail if enabled in a scenario where GPS is available but bad. Have you done any testing with the vertical velocity preflight check enabled in such a scenario? If yes, can you share the log please?

Reply:
Isn't the vertical velocity preflight check enabled by default? If not, how would I enable it?

Reply:
IMO: longer term the preflight checks shouldn't use innovations at all , but rather the covariances of the states. In the mean time this is a conservative check that can be performed, but I think we all agree it isn't optimal.